### ISSUE TITLE
Contract failed to verify. make: *** [deploy] Error 1

### ISSUE THREAD

**Issue by ThomasHeim11** [foundry-rs/foundry#5301]
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

 forge 0.2.0 (bf56869 2023-07-05T00:05:02.811991000Z)

### What command(s) is the bug in?

 make deploy ARGS="--network sepolia"

### Operating System

macOS (Apple Silicon)

### Describe the bug

I tried to deploy and verify multiple smart contracts using `make deploy ARGS="--network sepolia` and i got this error:

```
Start verification for (3) contracts
Start verifying contract `0x5d1bf24d8cd4d0e6ae7223df6815fefa8daa8da7` deployed on sepolia

Submitting verification for [src/libraries/OracleLib.sol:OracleLib] "0x5d1Bf24d8cd4d0e6AE7223DF6815FEfA8daA8dA7".
Submitted contract for verification:
        Response: `OK`
        GUID: `z9w3me7mt8pdvwqt6gbxaes6ygqcpcqf9hve3gi3irykcj75dy`
        URL:
        https://sepolia.etherscan.io/address/0x5d1bf24d8cd4d0e6ae7223df6815fefa8daa8da7
Contract verification status:
Response: `NOTOK`
Details: `Pending in queue`
Contract verification status:
Response: `NOTOK`
Details: `Other Exception - Please contact us for more information.`
Contract failed to verify.
make: *** [deploy] Error 1
```
I had this issue 2 weeks ago and I resolved it by updating `Foundryup` but it looks like the same error is back. I have tried to update both Foundry and Foundryup but it did not solve the issue. 

Her are my setup
foundry.tolm : 

```
[profile.default]
src = "src"
out = "out"
libs = ["lib"]
remappings = ['@chainlink/contracts/=lib/chainlink-brownie-contracts/contracts/', '@openzeppelin/contracts=lib/openzeppelin-contracts/contracts']

[etherscan]
mainnet = { key = "${ETHERSCAN_API_KEY}" }
sepolia = {key = "${ETHERSCAN_API_KEY}"}

[rpc_endpoints]
sepolia = "${SEPOLIA_RPC_URL}"

[invariant]
runs = 128
depth = 128
fail_on_revert = true

```

Makefile:

```
-include .env

.PHONY: all test clean deploy fund help install snapshot format anvil 

DEFAULT_ANVIL_KEY := 0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80

help:
	@echo "Usage:"
	@echo "  make deploy [ARGS=...]\n    example: make deploy ARGS=\"--network sepolia\""
	@echo ""
	@echo "  make fund [ARGS=...]\n    example: make deploy ARGS=\"--network sepolia\""

all: clean remove install update build

# Clean the repo
clean  :; forge clean

# Remove modules
remove :; rm -rf .gitmodules && rm -rf .git/modules/* && rm -rf lib && touch .gitmodules && git add . && git commit -m "modules"

install :; forge install chainaccelorg/foundry-devops@0.0.11 --no-commit && forge install smartcontractkit/chainlink-brownie-contracts@0.6.1 --no-commit && forge install foundry-rs/forge-std@v1.5.3 --no-commit && forge install openzeppelin/openzeppelin-contracts@v4.8.3 --no-commit

# Update Dependencies
update:; forge update

build:; forge build

test :; forge test 

coverage :; forge coverage --report debug > coverage-report.txt

snapshot :; forge snapshot

format :; forge fmt

anvil :; anvil -m 'test test test test test test test test test test test junk' --steps-tracing --block-time 1

NETWORK_ARGS := --rpc-url http://localhost:8545 --private-key $(DEFAULT_ANVIL_KEY) --broadcast

ifeq ($(findstring --network sepolia,$(ARGS)),--network sepolia)
	NETWORK_ARGS := --rpc-url $(SEPOLIA_RPC_URL) --private-key $(PRIVATE_KEY) --broadcast --verify --etherscan-api-key $(ETHERSCAN_API_KEY) -vvvv
endif

deploy:
	@forge script script/DeploySCD.s.sol:DeploySCD $(NETWORK_ARGS)

```

**Comment by Evalir** [comment]
Could you provide a minimal repro that I can test? The issue seems to involve libraries. having something i could test would be useful.

**Comment by ThomasHeim11** [comment]
Her is the repo for the contracts I tried to verify: https://github.com/ThomasHeim11/StableCryptoDollar

**Comment by ThomasHeim11** [comment]
I waited 3 days  and updated `foundry` again. Now it suddenly worked to verify the contracts.

**Comment by ungureanuu** [comment]
yes, confirming the issue but my fix was to upgrade solidity from 0.8.19 to 0.8.21

**Comment by deed3e** [comment]
![Screen Shot 2023-11-03 at 16 46 45](https://github.com/foundry-rs/foundry/assets/50208354/6472ff92-c854-432b-8996-1d67f0ba6308)
forge 0.2.0 (8bbde62 2023-11-03T09:29:47.566294Z)
So, i don't verify that
